src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -572,6 +572,14 @@ case class IndexLogEntry(
     tags.remove((plan, tag))
   }
 
+  def withCachedTag[T](plan: LogicalPlan, tag: IndexLogEntryTag[T])(
+    f: () => T): T = {
```

Comment:
Let's actually use the following so that the caller doesn't have to specify `() =>` (sorry for the wrong suggestion):
```suggestion
  def withCachedTag[T](plan: LogicalPlan, tag: IndexLogEntryTag[T])(f: => T): T = {
```